src/Apps/Artist/routes.tsx

```diff
@@ -87,42 +87,56 @@ export const routes: RouteConfig[] = [
     },
     query: graphql`
       query routes_ArtistTopLevelQuery($artistID: String!) @raw_response_type {
+        me {
+          id
+        }
         artist(id: $artistID) @principalField {
           ...ArtistApp_artist
           ...routes_Artist @relay(mask: false)
         }
       }
     `,
     render: ({ Component, props, match }) => {
-      if (Component && props) {
-        const { artist } = props as any
+      if (!(Component && props)) {
+        return null
+      }
 
-        if (!artist) {
-          return undefined
-        }
+      const { artist, me: user } = props as any
+      const { pathname } = match.location
 
-        const showArtistInsights =
-          showMarketInsights(artist) ||
-          (artist.insights && artist.insights.length > 0)
-        const hasArtistContent = hasOverviewContent(artist)
+      if (!artist) {
+        return undefined
+      }
 
-        const alreadyAtWorksForSalePath = match.location.pathname.includes(
-          `${artist.slug}/works-for-sale`
-        )
+      const showArtistInsights =
+        showMarketInsights(artist) ||
+        (artist.insights && artist.insights.length > 0)
+      const hasArtistContent = hasOverviewContent(artist)
 
-        const canShowOverview = showArtistInsights || hasArtistContent
+      const alreadyAtWorksForSalePath = pathname.includes(
+        `${artist.slug}/works-for-sale`
+      )
 
-        if (!canShowOverview && !alreadyAtWorksForSalePath) {
-          throw new RedirectException(`/artist/${artist.slug}/works-for-sale`)
-        }
+      const canShowOverview = showArtistInsights || hasArtistContent
 
-        return <Component {...props} />
+      if (user && !pathname.includes("/overview")) {
```

Comment:
It would be nice if this was added verbatim as a multiline comment just above these conditionals: 

<img width="634" alt="Screen Shot 2020-04-01 at 11 39 53 AM" src="https://user-images.githubusercontent.com/236943/78174103-94ea4780-740d-11ea-9d8a-a7e5182d7845.png">

Reply:
Done 👍